src/main/scala/castalia/StubService.scala

```diff
@@ -1,49 +1,63 @@
 package castalia
 
 import akka.actor.ActorSystem
+import akka.http.scaladsl.model.StatusCodes._
 import akka.http.scaladsl.server.Directives._
 import akka.http.scaladsl.server.Route
+import akka.pattern.after
 import castalia.model.ResponseConfig
 
-class StubService(theStubsByEndpoints: StubConfigsByEndpoint)(implicit val system: ActorSystem) extends Routes {
-  protected val serviceName = "StubRoutes"
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
-  protected def stubsByEndpoints: StubConfigsByEndpoint = theStubsByEndpoints
+class StubService(stubsByEndpoints: StubConfigsByEndpoint)(implicit val system: ActorSystem) extends Routes {
+  protected val serviceName = "StubRoutes"
 
   protected lazy val dynamicStubRoutes = {
-    def createRoute(endpoint: String, responses: ResponsesByRequest): Route = pathPrefix(endpoint) {
-      post {
-            path("responses") {
-              entity(as[ResponseConfig]) {
-                responseConfig =>
-                  responses(responseConfig.id) = (responseConfig.httpStatusCode, responseConfig.response)
-                  complete("")
+    def createRoute(endpoint: String, responses: ResponsesByRequest): Route =
+      pathPrefix(endpoint) {
+        post {
+              path("responses") {
+                entity(as[ResponseConfig]) {
+                  rc =>
+                    responses(rc.id) = (rc.delay, rc.httpStatusCode, rc.response)
+                    complete("")
+                }
               }
+        } ~
+        get {
+          path(Segment) { id =>
+            val response: Option[StubResponse] = responses.get(id)
+
+            response match {
+              case Some((Some(delay), statusCode: StatusCode, optResponse: AnyJsonObject)) =>
+                optResponse match {
+                  case Some(content) => complete(
+                    after(delay.duration, system.scheduler)
+                    (Future(statusCode, content.toJson)))
+                  case _ => complete(
+                    after(delay.duration, system.scheduler)
+                    (Future(statusCode, "")))
+                }
+              case Some((None, statusCode: StatusCode, optResponse: AnyJsonObject)) =>
```

Comment:
Maybe like this:
```response match {
              case Some((optDelay, statusCode: StatusCode, optResponse: AnyJsonObject)) => {
                val jsonContent: JsValue = optResponse match {
                    case Some(content) => content.toJson
                    case _ => JsObject.empty }

                optDelay match {
                  case Some(delay) => complete(
                    after(delay.duration, system.scheduler)
                    (Future(statusCode, jsonContent)))
                  case None => complete(statusCode, jsonContent)
                }
              }

              case _ => complete(NotImplemented, "Unknown response")
            }
```
But then if the response is empty , we get "{}" instead of "", this fails one of the tests.
Not sure if it is more readable also...